### PR TITLE
feat: add a `publish.yml` for publishing packages to npmjs

### PR DESCRIPTION
**Pull request by rushilbg**
BTC-1336

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}
+          git checkout -b release-$(git rev-parse --short HEAD)
+          echo "email=${{ secrets.NPM_EMAIL }}" > .npmrc
+          echo "@bitgo:registry=https://registry.npmjs.org" >> .npmrc
+          echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}" >> .npmrc
+          echo "//registry.npmjs.org/:always-auth=true" >> .npmrc
+
+      - name: Install Packages
+        run: yarn install --with-frozen-lockfile
+
```
Comment (rushilbg) [inline .github/workflows/publish.yml:57]: Need to add a prepare-release step here that increments the versions.

Comment (rushilbg) [reply .github/workflows/publish.yml:57]: Not needed, lerna publish should handle it automatically, updating that command below to add couple more things.

**Review by OttoAllmendinger — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          profile: minimal
+          override: true
+
+      - name: Install wasm-pack
+        run: |
+          cargo install wasm-pack
```
Comment (louib-bitgo) [inline .github/workflows/publish.yml:34]: @rushilbg Ideally this dependency would be in a `Cargo.toml` file in order to lock the version and the associated checksum.

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  group: ${{ github.workflow }}-${{ github.ref }}
+
+jobs:
+  publish:
+    name: Publish Release
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
```
Comment (louib-bitgo) [inline .github/workflows/publish.yml:18]: `v4` of this action is available now

**Pull request merged**